bcipy/helpers/triggers.py

```diff
@@ -638,3 +639,185 @@ def read_triggers(triggers_file: TextIO) -> List[Tuple[str, str, float]]:
     for _, (name, trg_type, stamp) in enumerate(records):
         corrected.append((name, trg_type, float(stamp) + offset))
     return corrected
+
+
+class TriggerType(Enum):
+    """
+    Enum for the primary types of Triggers.
+    """
+
+    NONTARGET = "nontarget"
+    TARGET = "target"
+    FIXATION = "fixation"
+    PROMPT = "first_pres_target"
+
+    @classmethod
+    def list(cls):
+        """Returns all enum values as a list"""
+        return list(map(lambda c: c.value, cls))
+
+    def __str__(self):
+        return f'{self.value}'
+
+
+class Trigger:
+    """
+    Object that encompasses data for a single trigger instance.
+    """
+
+    def __init__(self,
+                 label: str,
+                 type: TriggerType,
+                 time: str):
```

Comment:
Shouldn't `time` be a `float`? Also, I don't know if it would be confusing to override the Python global for `type`, since `trigger.type` doesn't mean the same thing as `type(trigger)`.

Reply:
I think I initially decided to make `time` a `str` because I wasn't sure if it was going to cause problems down the line with casting time back and forth between `str` and `float`, looking back I guess it does make more sense as just a regular `float`. Do you think this will be much of an issue, for usability or otherwise?